spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -273,4 +284,35 @@ private static void listDirRecursively(
       return files.iterator();
     };
   }
+
+  @VisibleForTesting
+  static class PartitionAwareHiddenPathFilter implements PathFilter, Serializable {
+
+    private final Set<String> hiddenPathPartitionNames;
+
+    PartitionAwareHiddenPathFilter(Set<String> hiddenPathPartitionNames) {
+      this.hiddenPathPartitionNames = hiddenPathPartitionNames;
+    }
+
+    @Override
+    public boolean accept(Path path) {
+      boolean isHiddenPartitionPath = hiddenPathPartitionNames.stream().anyMatch(path.getName()::startsWith);
+      return isHiddenPartitionPath || HiddenPathFilter.get().accept(path);
+    }
+
+    static PathFilter build(Map<Integer, PartitionSpec> specs) {
+      Set<String> partitionNames = new HashSet<>();
```

Comment:
Nit: We don't use `new HashSet<>()` directly, as generally speaking that will usually lead to having to grow the underlying collection multiple times.

We use `Sets.newHashSet()`, or preferably the constructor that directly takes in its elements or the expected size if possible. Otherwise, we use `Sets.newHashSet()`.

We also shade and relocate Guava so that we don't conflict with the user's existing Guava version on their classpath. The correct import would be for `org.apache.iceberg.relocated.com.google.common.collect.Sets`.

All that said, I think that you might be able to apply a `filter` and then use `Collectors.toSet` below instead.